src/libs/SubscriptionUtils.ts

```diff
@@ -30,25 +143,168 @@ Onyx.connect({
     waitForCollectionCallback: true,
 });
 
-let ownerBillingGraceEndPeriod: OnyxEntry<number>;
-Onyx.connect({
-    key: ONYXKEYS.NVP_PRIVATE_OWNER_BILLING_GRACE_PERIOD_END,
-    callback: (value) => (ownerBillingGraceEndPeriod = value),
-});
-
-let amountOwed: OnyxEntry<number>;
-Onyx.connect({
-    key: ONYXKEYS.NVP_PRIVATE_AMOUNT_OWNED,
-    callback: (value) => (amountOwed = value),
-});
-
 let allPolicies: OnyxCollection<Policy>;
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.POLICY,
     callback: (value) => (allPolicies = value),
     waitForCollectionCallback: true,
 });
 
+function getOverdueGracePeriodDate(): number {
+    return billingGracePeriod;
+}
+
+function hasOverdueGracePeriod(): boolean {
+    return !!billingGracePeriod ?? false;
+}
+
+function hasGracePeriodOverdue(): boolean {
+    return !!billingGracePeriod && Date.now() < new Date(billingGracePeriod).getTime();
+}
+
+function getAmountOwed(): number {
+    return amountOwed ?? 0;
+}
+
+function hasCardAuthenticatedError() {
+    return stripeCustomerId?.status === 'authentication_required' && amountOwed === 0;
+}
+
+function hasBillingDisputePending() {
+    return !!billingDisputePending ?? false;
+}
+
+function hasCardExpiredError() {
+    return billingStatus?.declineReason === 'expired_card' && amountOwed !== 0;
+}
+
+function hasInsufficientFundsError() {
+    return billingStatus?.declineReason === 'insufficient_funds' && amountOwed !== 0;
+}
+
+function getCardForSubscriptionBilling(): Fund | undefined {
+    return Object.values(fundList ?? {}).find((card) => card?.isDefault);
+}
+
+function hasCardExpiringSoon(): boolean {
+    const card = getCardForSubscriptionBilling();
+
+    if (!card) {
+        return false;
+    }
+
+    return !billingStatus && card?.accountData?.cardMonth === new Date().getMonth() + 1;
```

Comment:
If the `!billingStatus` check is needed anyway then why not move it to the first line before calling `getCardForSubscriptionBilling` ?

Reply:
Right